client/components/domains/domain-product-price/index.jsx

```diff
@@ -43,10 +51,9 @@ var DomainProductPrice = React.createClass( {
 		} );
 	},
 	priceText() {
-		const selectedSite = sitesList.getSelectedSite();
 		if ( ! this.props.price ) {
 			return this.translate( 'Free' );
-		} else if ( domainsWithPlansOnlyTestEnabled && ! cartItems.isNextDomainFree( this.props.cart ) && ( ! selectedSite || ! isPlan( selectedSite.plan ) ) ) {
```

Comment:
You removed the `cartItems.isNextDomainFree` check here - I think it's important, because otherwise it causes this regression:
![2016-05-20-013743_643x520_scrot](https://cloud.githubusercontent.com/assets/3392497/15413276/e4de2902-1e2e-11e6-93c4-f9db6d5756ed.png)
Reproducing: add a plan to cart (but don't finish checkout), go back to Domain Search.